gobblin-modules/gobblin-orc/src/main/java/org/apache/gobblin/writer/GobblinBaseOrcWriter.java

```diff
@@ -258,7 +261,18 @@ public void close()
   public void commit()
       throws IOException {
     closeInternal();
+    if(this.validateORCDuringCommit) {
```

Comment:
https://github.com/apache/gobblin/pull/3818#discussion_r1379722450

This issue is still present. We want to move this to close function and not just commit because we flush the buffer there too and if it's malformed then we want to catch it.

Reply:
Im curious whats the validation and how it works. Does it validate on the header or something else.

Reply:
It does all sorts of validations. I attached below 1 example. You can dig around the class for a bunch of others
- https://github.com/apache/orc/blob/24beffb6fed6d408e25654e53c255f564c8bd8a9/java/core/src/java/org/apache/orc/impl/ReaderImpl.java#L797C1-L802

Reply:
https://github.com/apache/gobblin/blob/c865b6a8e4628f9602671151a1a33c8a27dacdef/gobblin-core/src/main/java/org/apache/gobblin/publisher/BaseDataPublisher.java#L492C6-L517

In case you would like to read more about how it's done at a directory level